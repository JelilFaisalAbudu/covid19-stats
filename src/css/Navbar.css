header {
  background-color: #282c34;
  position: sticky;
  top: 0;
  left: 0;
}

.nav {
  display: flex;
  justify-content: space-between;
  width: 100%;
  position: relative;
  padding: 1em 5em;
}

.nav > .nav-header {
  display: inline;
}

.nav > .nav-header > .nav-title {
  display: inline-block;
  font-size: 22px;
  color: #fff;
}

.nav > .nav-btn {
  display: none;
}

.nav .nav-links {
  font-size: 18px;
}

.nav > a {
  display: inline-block;
  text-decoration: none;
  color: #efefef;
}

.nav > .nav-links > a:hover {
  background-color: rgba(0, 0, 0, 0.3);
}

.nav > #nav-check {
  display: none;
}

@media (max-width: 600px) {
  .nav > .nav-btn {
    display: inline-block;
    position: absolute;
    right: 0;
    top: 0;
  }

  .nav > .nav-btn > label {
    display: inline-block;
    width: 50px;
    height: 50px;
    padding: 13px;
  }

  .nav > .nav-btn > label:hover,
  .nav #nav-check:checked ~ .nav-btn > label {
    background-color: rgba(0, 0, 0, 0.3);
  }

  .nav > .nav-btn > label > span {
    display: block;
    width: 25px;
    height: 10px;
    border-top: 2px solid #eee;
  }

  .nav > .nav-links {
    position: absolute;
    display: block;
    width: 100%;
    background-color: #333;
    height: 0;
    transition: all 0.3s ease-in;
    overflow-y: hidden;
    top: 50px;
    left: 0;
  }

  .nav > .nav-links > a {
    display: block;
    width: 100%;
  }

  .nav > #nav-check:not(:checked) ~ .nav-links {
    height: 0;
  }

  .nav > #nav-check:checked ~ .nav-links {
    height: calc(100vh - 50px);
    overflow-y: auto;
  }
}
