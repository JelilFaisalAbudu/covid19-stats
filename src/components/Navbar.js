/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import { Link } from 'react-router-dom';
import '../css/Navbar.css';

const Navbar = () => (
  <nav className="nav">
    <input type="checkbox" id="nav-check" />
    <div className="nav-header">
      <div className="nav-title">
        <Link className="brand" to="/">WORLD COVID-19 STATISTICS</Link>
      </div>
    </div>
    <div className="nav-btn">
      <label htmlFor="nav-check">
        <span />
        <span />
        <span />
      </label>
    </div>

    <div className="nav-links">
      <Link to="/">Home</Link>
    </div>
  </nav>
);

export default Navbar;
