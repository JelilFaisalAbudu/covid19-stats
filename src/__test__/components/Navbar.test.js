import { shallow } from 'enzyme';
import '../setup/setupTests';
import Navbar from '../../components/Navbar';

const setup = () => shallow(<Navbar />);

describe('Navbar component', () => {
  let component;

  beforeEach(() => {
    component = setup();
  });

  it('finds the only 1 nav', () => {
    const element = component.find('.nav');
    expect(element.length).toEqual(1);
  });

  test('renders the nav in the page', () => {
    const item = component.find('.nav');
    expect(item).toMatchSnapshot();
  });

  test('renders the nav brand in the page', () => {
    const item = component.find('.brand');
    expect(item).toMatchSnapshot();
  });

  test('renders the navBrand in the page', () => {
    const item = component.find('.nav-links');
    expect(item).toMatchSnapshot();
  });
});
