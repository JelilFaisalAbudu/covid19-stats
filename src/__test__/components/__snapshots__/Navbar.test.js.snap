// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders the nav brand in the page 1`] = `ShallowWrapper {}`;

exports[`Navbar component renders the nav in the page 1`] = `ShallowWrapper {}`;

exports[`Navbar component renders the navBrand in the page 1`] = `ShallowWrapper {}`;
